common/sai.py

```diff
@@ -614,14 +614,21 @@ def apply_rec(self, fname):
                     # Update OIDs in the key
                     key = self.__update_key(rec[0], key)
                     # Convert into ["sai-object-type", "key"]
-                    key = key.split(":", 1)[1]
+                    if ":" in key:
+                        key = key.split(":", 1)[1]
 
-                    if key.startswith("{"):
+                    if type(key) is dict or key.startswith("{"):
                         key = json.loads(key)
                     bulk_keys.append(key)
                     bulk_attrs.append(attrs)
 
-                self.bulk_create(record[0][1], bulk_keys, bulk_attrs)
+                if type(key) is dict or key.startswith("{"):
+                    self.bulk_create(record[0][1], bulk_keys, bulk_attrs)
+                else:
+                    _, new_keys, _ = self.bulk_create(record[0][1], None, bulk_attrs, len(record)-1)
+                    for idx in range(0, len(new_keys)):
+                        if "oid:" in new_keys[idx]:
+                            self.rec2vid[new_keys[idx]] = new_keys[idx]
```

Comment:
I think something like `self.rec2vid[record[idx + 1][0]]` should be instead of `self.rec2vid[new_keys[idx]]`. Please check.